### ISSUE TITLE
Sparkling eyes.

### ISSUE THREAD

**Issue by zekageri** [philbowles/ESPAsyncWebServer#4]
My eyes sparkled when I saw this magical library.
Unfortunately, I soon had to realize that I could not use it without the websocket implementation. :( 
Is there any chance that this functionality will be implemented?

**Comment by HamzaHajeir** [comment]
Unfortunately, The author (maintainer) Philbowles has passes away on June 2022.

You can either check the original [ESPAsyncWebServer](https://github.com/me-no-dev/ESPAsyncWebServer), or Phil's introduced [H4AsyncWebServer](https://github.com/philbowles/H4AsyncWebServer), he made an implementation of Websockets. there, but no authentication is present.

**Comment by zekageri** [comment]
Thank you for the information. Iam really sorry.

**Comment by HamzaHajeir** [comment]
No problem, @zekageri.

You can keep an eye on my forks of H4 libraries, if any update occurs.

**Comment by zekageri** [comment]
Yeah, I did watch h4 libs. Unfortunately it can not be integrated to an existing system. It sure looks promising but incompatible with my project. But I will build a project with it one day.

**Comment by zekageri** [comment]
@HamzaHajeir So you picked up all h4 lib? Including the ESPAsyncWebServer and the Async tcp too? Are they reliable?

**Comment by HamzaHajeir** [comment]
I've utilized H4AsyncWebServer in a project, I've not faced errors with it until now xD.

For H4AsyncTCP and H4AsyncMQTT, I'm in debugging and fixing them to reach a reliable level. And perhaps upgrading them (TLS and MQTT 5.0).

At their current state, these are honestly more reliable than the previous ones. Would be more robust after publishin the updates.

**Comment by zekageri** [comment]
The thing is that i desperately searching for a fork of `AsyncTCP` and `AsyncWebServer` for half a year now. All of the versions and forks has the same bug. They can't handle concurrent client connections and they all crash at some point. (always with the same stack trace pointing to the libs ) With `H4AsyncWebserver` and `H4AsyncTCP` i must rebuild my projects from the ground up because they have a very different logic.

**Comment by HamzaHajeir** [comment]
I see, I've already tested (with limited time) the functionality of both MQTT and Webserver combined, they function as expected.

Just AsyncWebServer might not work if you've enabled H4AT_DEBUG level to a high one, perhaps to timing.

Yes, H4AsyncWebserver has different logic, to me I was going to work on ESPAsync. one, but it didn't work and decided later to fully convert to H4*s.

At some point, you might make a small sketch combining H4 libraries and validate reliability before deciding whether to change or not.

**Comment by zekageri** [comment]
The problem is that I afraid to make a project at this point. If you test the normal libs, they work as expected as long as you keep your endpoints below a few and minimal. As soon as you build a bigger project with more endpoints and events and more then 3-4 client socket connections they start to fail. Iam afraid of minimal tests. But I will definitely try it

**Comment by HamzaHajeir** [comment]
I see, you might simulate the situation by multiple instances in that
simple sketch.


If you have a MVCE replicates the crashes by those libraries, you can
convert them into H4 ones with minimum efforts, then test.

**Comment by HamzaHajeir** [comment]
Hi @zekageri 

You can checkout the stable H4AsyncWebServer, use [my fork](https://github.com/HamzaHajeir/H4AsyncWebServer), a [complete PlatformIO environment for H4Plugins is available](https://github.com/HamzaHajeir/H4Plugins_Env).

You might check dependencies under platformio.ini.

**Comment by zekageri** [comment]
So the license prevents the commercial use?

**Comment by HamzaHajeir** [comment]
Yeah unfortunately, it's an upcoming issue to resolve, I think we'll start
a discussion soon for this particular issue. You can join the Facebook
group to participate and get updates.

**Comment by HamzaHajeir** [comment]
This one:
https://www.facebook.com/groups/444344099599131/

**Comment by zekageri** [comment]
I think i have applied to this group but never got accepted. My name is Zékány Gergő on facebook

**Comment by zekageri** [comment]
I have just reapplied

**Comment by HamzaHajeir** [comment]
I see, it's unfortunate that he is the only admin ... It's another issue. I think many people are in the waiting list.

Can you send me your account link, so at least if we made a group chat or another group you can get in.

**Comment by zekageri** [comment]
Yes, no problem.

https://www.facebook.com/gergo.zekany.39/

**Comment by HamzaHajeir** [comment]
Checkout the request

‫في الاثنين، 10 يوليو 2023 في 2:42 م تمت كتابة ما يلي بواسطة ‪DrRandom‬‏ <‪
***@***.***‬‏>:‬
